superset-frontend/src/explore/components/controls/AnnotationLayerControl/index.tsx

```diff
@@ -36,30 +49,64 @@ const AnnotationLayer = AsyncEsmComponent(
   () => <div style={{ width: 450, height: 368 }} />,
 );
 
-const propTypes = {
-  colorScheme: PropTypes.string.isRequired,
-  annotationError: PropTypes.object,
-  annotationQuery: PropTypes.object,
-  vizType: PropTypes.string,
+export interface Annotation {
+  name: string;
+  show?: boolean;
+  annotation: any;
+  timeout: any;
+  key: any;
+  formData?: any;
+  isDashboardRequest?: boolean;
+  force?: boolean;
+}
 
-  validationErrors: PropTypes.array,
-  name: PropTypes.string.isRequired,
-  actions: PropTypes.object,
-  value: PropTypes.arrayOf(PropTypes.object),
-  onChange: PropTypes.func,
-  refreshAnnotationData: PropTypes.func,
-};
+export interface Props {
+  colorScheme: string;
+  annotationError: Record<string, string>;
+  annotationQuery: Record<string, AbortController>;
+  vizType: string;
+  validationErrors: JsonObject[];
+  name: string;
+  actions: {
+    setControlValue: HandlerFunction;
+  };
+  value: Annotation[];
+  onChange: (annotations: Annotation[]) => void;
+  refreshAnnotationData: (payload: Payload) => void;
+  theme: SupersetTheme;
+}
 
+export interface PopoverState {
+  popoverVisible: Record<number | string, boolean>;
+  addedAnnotationIndex: number | null;
+}
+export interface Explore {
+  can_add: boolean;
+  can_download: boolean;
+  can_overwrite: boolean;
+  isDatasourceMetaLoading: boolean;
+  isStarred: boolean;
+  triggerRender: boolean;
+  datasource: Dataset;
+  controls: ControlStateMapping;
+  form_data: QueryFormData;
+  slice: Slice;
+  controlsTransferred: string[];
+  standalone: boolean;
+  force: boolean;
+  common: JsonObject;
+}
```

Comment:
Please reuse the `ExplorePageState['explore']` type in `src/explore/types' to minimize the maintainance.

Reply:
done